apps/aechannel/src/aesc_chain_watcher.erl

```diff
@@ -383,59 +531,81 @@ check_requests(Reqs, St, C) ->
 %% Trap exceptions INSIDE the db activity for meaningful error reporting
 check_requests(Reqs, St, C, []) ->
     Reqs1 = reset_if_fork_switch(Reqs, C),
-    try check_requests_(Reqs1, St, C, [])
+    try check_requests_(Reqs1, St, C)
     ?_catch_(error, E, StackTrace)
         lager:error("CAUGHT ~p / ~p", [E, StackTrace]),
+        lager:error("Smaller trace ~p", [compact_trace(StackTrace)]),
         error(E)
     end.
 
+compact_trace([{M,F,As,Info}|T]) when is_list(As) ->
+    [{M,F,length(As),Info}|compact_trace(T)];
+compact_trace([H|T]) ->
+    [H|compact_trace(T)];
+compact_trace([]) ->
+    [].
+
+
 %% If there's a fork switch, don't assume channel state or depth calculations
 reset_if_fork_switch(Reqs, #{scenario := S}) when ?IS_SCENARIO(S) ->
     case S of
         fork_switch ->
-            [maps:remove(check_at_height,
-                         maps:remove(locked_until, R)) || R <- Reqs];
+            reset_after_fork_switch(Reqs);
         _ ->
             Reqs
     end.
 
-init_cache(top, St) ->
-    C0 = init_cache_(St),
-    %% Set only top_hash. Further top_info fetched on-demand
-    {_THash, C1} = top_hash(C0),
-    C1;
+reset_after_fork_switch(Reqs) when is_list(Reqs) ->
```

Comment:
So this function returns: `list(#req{}) | function()`?

Reply:
Yeah, the assumption was that if a fork switch is detected, all requests must be checked, and that this would be better done continuation-style. Given the likelihood of microblock forks, this is probably a bad assumption to begin with. It also makes certain optimizations harder, so I've been leaning towards removing the funs.